ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ServerInfoImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.ServerInfo;
+import java.util.Objects;
+
+public class ServerInfoImpl implements ServerInfo {
+  private final String serverVersion;
+  private final String kafkaClusterId;
+  private final String ksqlServiceId;
+
+  public ServerInfoImpl(
+      final String serverVersion,
+      final String kafkaClusterId,
+      final String ksqlServiceId
+  ) {
+    this.serverVersion = Objects.requireNonNull(serverVersion);
```

Comment:
nit: use second argument in `Objects.requireNonNull()` to produce a more informative error message if something goes wrong here.
```suggestion
    this.serverVersion = Objects.requireNonNull(serverVersion. "serverVersion");
```
and similar for the others.